### ISSUE TITLE
Price filter with including tax (In Backend Catalog Prices set Excluding tax) in magneto 2.2.6

### ISSUE THREAD

**Issue by karangar** [magento/magento2#18547]
### Preconditions

1.I have set Backend Catalog Price as a Excluding Tax  And Front side Display Product Price in Catalog set as Including Tax.
2. Magento 2.2.6

### Steps to reproduce

1. Magento 2.2.6
2. Go to Backend and set this Tax Calculation configuration 
     Stores -> Configuration -> Sales -> Tax 
     
![screenshot-192 168 10 225-2018 10 12-12-04-20](https://user-images.githubusercontent.com/34768419/46853224-19609900-ce1b-11e8-9d9a-c7043df60132.png)
    
    Stores -> Taxes -> Tax Zones and Rates
![screenshot-192 168 10 225-2018 10 12-12-07-32](https://user-images.githubusercontent.com/34768419/46853275-3b5a1b80-ce1b-11e8-8b46-79b159ef5de4.png)

Stores -> Taxes ->  Tax Rules
![screenshot-192 168 10 225-2018 10 12-12-12-27](https://user-images.githubusercontent.com/34768419/46853330-562c9000-ce1b-11e8-8add-1537e0066fdd.png)

3. Go to Front end listing page and filter it by Price  Range 40 to 50

### Expected result

1. In listing Page After Price filter Display result between  40 to 50

### Actual result

1. In listing Page After Price filter Display result in some products are displayed greater than 50. Please check below screen shot .
![screenshot-192 168 10 225-2018 10 12-12-17-26](https://user-images.githubusercontent.com/34768419/46853576-2e89f780-ce1c-11e8-89ca-d7265d07baf3.png)

**Comment by magento-engcom-team** [comment]
Hi @karangar. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@karangar do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by karangar** [comment]
Hello @magento-engcom-team 
Please Give me 2.2.6 sample data instance

**Comment by karangar** [comment]
@magento-engcom-team give me 2.2.6 instance

**Comment by magento-engcom-team** [comment]
Hi @karangar. Thank you for your request. I'm working on Magento 2.2.6 instance for you

**Comment by magento-engcom-team** [comment]
Hi @karangar, here is your [Magento instance](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18547-2-2-6/).
Admin access: [http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18547-2-2-6//admin](http://ec2-34-228-235-121.compute-1.amazonaws.com/i-18547-2-2-6//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by karangar** [comment]
How can I indexing for this instance?

**Comment by karangar** [comment]
Can you please do needful to run indexing for this instance.
I have uploaded few products to show different products at frontend.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-andrii. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-andrii Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96615`, `MAGETWO-96616` were created

**Comment by karangar** [comment]
#MM19IN

**Comment by magento-engcom-team** [comment]
@karangar, thank you for your report.     
    
Unfortunately, we are archiving this ticket now as it did not get much attention from both Magento Community and Core developers for an extended period. This is done in an effort to create a quality, community-driven backlog which will allow us to allocate the required attention more easily.

Please feel free to comment, reopen or create new ticket according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) 
if you are still facing this issue on the latest `2.3-develop` branch. Thank you for collaboration.

**Comment by maderlock** [comment]
In an effort to focus on quality by allocating attention, can I please point out that we are still facing this on 2.3 and feel that closing this ticket will hide the fact that this issue still exists, so others will find it harder to find this information.

**Comment by maderlock** [comment]
Still seeing this on 2.3.3pt2. Looking to upgrade to 2.3.5. I don't suppose this has been looked at in the meantime?

**Comment by barbazul** [comment]
I can confirm this issue still exists in Magento 2.4.1
